test/run_test

```diff
@@ -46,18 +51,22 @@ function cleanup() {
     echo "Done."
   done
   rmdir $CIDFILE_DIR
+
+  ct_path_foreach "$volumes_to_clean" cleanup_volume_dir
 }
 trap cleanup EXIT
 
-function cleanup_volume() {
-  local volume_dir="$1"; shift
-  local cid="$1"; shift
-  local volume_args="-v ${volume_dir}:/var/lib/pgsql/data:Z"
-
-  docker stop $cid
-  local rm_cmd='/bin/rm -rf /var/lib/pgsql/data/userdata'
-  docker run $volume_args --rm $IMAGE_NAME /bin/sh -c "$rm_cmd"
-  rmdir ${volume_dir}
+cleanup_volume_dir ()
+{
+  test ! -d "$1" && : "WARN: cleaned $1 for some reason" && return 0
+  # When we run this test script as non-root (we should?), the PostgreSQL server
+  # within container is still run under 'postgres' user.  It means that, taking
+  # into account 0077 umask of PostgreSQL server, we are unable to remove files
+  # created by server.  That's why we need to let docker escalate the privileges
+  # again.
+  local datadir=/var/lib/pgsql/data
+  docker run -v "$1:$datadir:Z" --rm "$IMAGE_NAME" /bin/sh -c "/bin/rm -rf $datadir/userdata"
```

Comment:
I thought that container is also tested with "random" user (`"-u 12345"`). I tried this manually and in that case this does not work for me. So I had to run cleaning with `-u 0`.

But tests are passing, so maybe I'm doing something wrong. @praiskup Or random uid is not tested with mounted volume?